From b190e81bfe66dbc353038106f3f6a166ae2b0f18 Mon Sep 17 00:00:00 2001
From: WingMan Kwok <w-kwok2@ti.com>
Date: Thu, 12 Nov 2015 16:12:50 -0500
Subject: [PATCH 05/23] ARM: keystone: dts: add gbe and 10gbe netcp cpts
 bindings

This patch adds the netcp cpts DT bindings for the
gbe and 10gbe devices on the Keystone2 platforms.

Signed-off-by: WingMan Kwok <w-kwok2@ti.com>
---
 arch/arm/boot/dts/k2e-netcp.dtsi  | 28 +++++++++++++++++++++++++---
 arch/arm/boot/dts/k2hk-netcp.dtsi | 30 ++++++++++++++++++++++++++----
 arch/arm/boot/dts/k2l-netcp.dtsi  | 13 ++++++++++++-
 3 files changed, 63 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/k2e-netcp.dtsi b/arch/arm/boot/dts/k2e-netcp.dtsi
index 320cf4d..dd1a964 100644
--- a/arch/arm/boot/dts/k2e-netcp.dtsi
+++ b/arch/arm/boot/dts/k2e-netcp.dtsi
@@ -222,7 +222,7 @@ netcp: netcp@24000000 {
 	ranges = <0 0x24000000 0x1000000>;
 
 	clocks = <&clkpa>, <&clkcpgmac>, <&chipclk12>;
-	clock-names = "pa_clk", "ethss_clk", "cpts_rft_clk";
+	clock-names = "pa_clk", "ethss_clk", "cpts";
 	dma-coherent;
 
 	ti,navigator-dmas =	<&dma_gbe 0>,
@@ -279,6 +279,17 @@ netcp: netcp@24000000 {
 			tx-channel = "nettx";
 			phys = <&gbe_serdes0>, <&gbe_serdes1>;
 
+			cpts {
+				status = "ok";
+				rftclk_sel = <0>;
+				/*rftclk_freq = <625000000>;*/
+				clock_mult = <4096>;
+				clock_shift = <9>;
+				clock_div = <5>;
+				ts_comp_length = <3>;
+				ts_comp_polarity = <1>; /* 1 - assert high */
+			};
+
 			interfaces {
 				gbe0: interface-0 {
 					slave-port = <0>;
@@ -471,8 +482,8 @@ netcpx: netcpx@2f00000 {
 	#size-cells	= <1>;
 	ranges;
 
-	clocks		= <&clkxge>;
-	clock-names	= "clk_xge";
+	clocks		= <&clkxge>, <&chipclk12>;
+	clock-names	= "clk_xge", "cpts";
 	dma-coherent;
 	/* big-endian; */
 
@@ -496,6 +507,17 @@ netcpx: netcpx@2f00000 {
 			tx-channel = "xnettx";
 			phys = <&xgbe_serdes>;
 
+			cpts {
+				status = "ok";
+				rftclk_sel = <0>;
+				/*rftclk_freq = <625000000>;*/
+				/*clock_mult = <2560>;*/
+				/*clock_shift = <10>;*/
+				/*clock_div = <1>;*/
+				ts_comp_length = <3>;
+				ts_comp_polarity = <1>; /* 1 - assert high */
+			};
+
 			interfaces {
 				xgbe0: interface-0 {
 					slave-port = <0>;
diff --git a/arch/arm/boot/dts/k2hk-netcp.dtsi b/arch/arm/boot/dts/k2hk-netcp.dtsi
index 9701838..3400a87 100644
--- a/arch/arm/boot/dts/k2hk-netcp.dtsi
+++ b/arch/arm/boot/dts/k2hk-netcp.dtsi
@@ -202,8 +202,8 @@ netcp: netcp@2000000 {
 	/* NetCP address range */
 	ranges  = <0 0x2000000 0x100000>;
 
-	clocks = <&clkpa>, <&clkcpgmac>, <&chipclk12>;
-	clock-names = "pa_clk", "ethss_clk", "cpts_rft_clk"	;
+	clocks = <&clkpa>, <&clkcpgmac>, <&chipclk13>;
+	clock-names = "pa_clk", "ethss_clk", "cpts";
 	dma-coherent;
 
 	ti,navigator-dmas =	<&dma_gbe 22>,	/* rx flow for CPSW port 0 */
@@ -245,6 +245,17 @@ netcp: netcp@2000000 {
 			tx-channel = "nettx";
 			phys = <&gbe_serdes>;
 
+			cpts {
+				status = "ok";
+				rftclk_sel = <1>;
+				/*rftclk_freq = <400000000>;*/
+				clock_mult = <2560>;
+				clock_shift = <10>;
+				clock_div = <1>;
+				ts_comp_length = <3>;
+				ts_comp_polarity = <1>; /* 1 - assert high */
+			};
+
 			interfaces {
 				gbe0: interface-0 {
 					slave-port = <0>;
@@ -422,8 +433,8 @@ netcpx: netcp@2f00000 {
 	#size-cells	= <1>;
 	ranges;
 
-	clocks		= <&clkxge>;
-	clock-names	= "clk_xge";
+	clocks		= <&clkxge>, <&chipclk13>;
+	clock-names	= "clk_xge", "cpts";
 	dma-coherent;
 	/* big-endian; */
 
@@ -447,6 +458,17 @@ netcpx: netcp@2f00000 {
 			tx-channel = "xnettx";
 			phys = <&xgbe_serdes>;
 
+			cpts {
+				status = "ok";
+				rftclk_sel = <1>;
+				/*rftclk_freq = <400000000>;*/
+				/*clock_mult = <2560>;*/
+				/*clock_shift = <10>;*/
+				/*clock_div = <1>;*/
+				ts_comp_length = <3>;
+				ts_comp_polarity = <1>; /* 1 - assert high */
+			};
+
 			interfaces {
 				xgbe0: interface-0 {
 					slave-port = <0>;
diff --git a/arch/arm/boot/dts/k2l-netcp.dtsi b/arch/arm/boot/dts/k2l-netcp.dtsi
index bb57002..5d702e5 100644
--- a/arch/arm/boot/dts/k2l-netcp.dtsi
+++ b/arch/arm/boot/dts/k2l-netcp.dtsi
@@ -207,7 +207,7 @@ netcp: netcp@26000000 {
 	ranges = <0 0x26000000 0x1000000>;
 
 	clocks = <&clkpa>, <&clkcpgmac>, <&chipclk12>, <&clkosr>;
-	clock-names = "pa_clk", "ethss_clk", "cpts_rft_clk", "osr_clk";
+	clock-names = "pa_clk", "ethss_clk", "cpts", "osr_clk";
 	dma-coherent;
 
 	ti,navigator-dmas =	<&dma_gbe 0>,
@@ -264,6 +264,17 @@ netcp: netcp@26000000 {
 			tx-channel = "nettx";
 			phys = <&gbe_serdes0>, <&gbe_serdes1>;
 
+			cpts {
+				status = "ok";
+				rftclk_sel = <0>;
+				/*rftclk_freq = <399360000>;*/
+				/*clock_mult = <5000>;*/
+				/*clock_shift = <10>;*/
+				/*clock_div = <3>; */
+				ts_comp_length = <3>;
+				ts_comp_polarity = <1>; /* 1 - assert high */
+			};
+
 			interfaces {
 				gbe0: interface-0 {
 					slave-port = <0>;
-- 
1.9.1

